macros/union_data.sql

```diff
@@ -116,13 +118,14 @@
         {%- set relation.value=adapter.get_relation(
             database=source(default_schema, table_identifier).database,
             schema=source(default_schema, table_identifier).schema,
-            identifier=var(identifier_var, table_identifier)
+            identifier=connector_table_name_override if (connector_table_name_override and not var('use_table_name_identifer_override', false)) else var(identifier_var, connector_table_name_override if connector_table_name_override else table_identifier)
```

Comment:
Sure, let's take the account_type table in Netsuite2. The [`source`](https://github.com/fivetran/dbt_netsuite_source/blob/feature/add-union-data/models/netsuite2/src_netsuite2.yml#L16) table name is `account_type`, while the actual connector table name is `accounttype`. 

In the [_tmp](https://github.com/fivetran/dbt_netsuite_source/blob/feature/add-union-data/models/netsuite2/tmp/stg_netsuite2__account_types_tmp.sql) model, we provide `accounttype` as the `connector_table_name_override` instead of the typical `table_identifier`, which would be `account_type` in this case. we have not configured the `use_table_name_identifier_override` variable, so that remains false and the macro uses the `connector_table_name_override`. For tables that don't have special/wonky connector names, the macro uses the default `table_identifier` (example: [accounts](https://github.com/fivetran/dbt_netsuite_source/blob/feature/add-union-data/models/netsuite2/tmp/stg_netsuite2__accounts_tmp.sql))

Back to account_types -- in our integration tests, the seed file is named `netsuite2_account_type_data`. We use the `netsuite2_account_type_identifier` variable to point to this name [here](https://github.com/fivetran/dbt_netsuite_source/blob/feature/add-union-data/integration_tests/dbt_project.yml#L49). However, because the respective _tmp model has a `connector_table_name_override`, i was seeing the `no ACCOUNT_TYPE table found in your schema. Fivetran will create an empty staging model yadda yadda` log message for this and every table for which we used `connector_table_name_override`.

To ensure that we _could_ use identifier variables even for these weird tables, we now set the global `use_table_name_identifier_override` variable to True in the integration_tests/dbt_project.yml file. This means that we'll use the `connector_table_name_override` if it's provided, but _only_ if we're not overriding the table identifier completely with our own custom identifier (in this case `netsuite2_account_type_data`). 

But let's say that our seed file for `accounting_book_subsidiaries` aligns with the actual connector table name of `accountingbooksubsidiaries` (which we have configured in the [_tmp](https://github.com/fivetran/dbt_netsuite_source/blob/feature/add-union-data/models/netsuite2/tmp/stg_netsuite2__accounting_book_subsidiaries_tmp.sql) model as well), so it's `accountingbooksubsidiaries.csv`. 

The last part of the highlighted line, `var(identifier_var, connector_table_name_override if connector_table_name_override else table_identifier)` makes it so that we can use identifier variables for some tables but we don't _need_ to use them for all of the tables (despite our globally-defined variable). The package will pick up `accountingbooksubsidiaries` for the respective staging model and still pick up `netsuite2_account_type_data` for the account_type staging model. This would work as well if we had a seed file called `accounts.csv` for the accounts table, which doesn't use the `connector_table_name_override`.